driver-core/src/test/unit/com/mongodb/internal/connection/InitialDnsSeedListDiscoveryProseTest.java

```diff
@@ -0,0 +1,147 @@
+            cluster.close();
+        }
+    }
+
+    @ParameterizedTest(name = "mongodb+srv://{0} => {1}")
+    @CsvSource({
+            "localhost, mongo.localhost",
+            "mongo.local, driver.mongo.local"
+    })
+    @DisplayName("1. Allow SRVs with fewer than 3 '.' separated parts")
+    void test1(final String srvHost, final String resolvedHost) {
+        doTest(srvHost, resolvedHost, false);
+    }
+
+    @ParameterizedTest(name = "mongodb+srv://{0} => {1}")
+    @CsvSource({
+            "localhost, localhost.mongodb",
+            "mongo.local, test_1.evil.local",
+            "blogs.mongodb.com, blogs.evil.com"
+    })
+    @DisplayName("2. Throw when return address does not end with SRV domain")
+    void test2(final String srvHost, final String resolvedHost) {
+        doTest(srvHost, resolvedHost, true);
+    }
+
+    @ParameterizedTest(name = "mongodb+srv://{0} => {1}")
+    @CsvSource({
+            "localhost, localhost",
+            "mongo.local, mongo.local"
+    })
+    @DisplayName("3. Throw when return address is identical to SRV hostname and return address does not contain '.' separating shared part of domain")
+    void test3(final String srvHost, final String resolvedHost) {
+        doTest(srvHost, resolvedHost, true);
+    }
+
+    @ParameterizedTest(name = "mongodb+srv://{0} => {1}")
+    @CsvSource({
+            "localhost, test_1.cluster_1localhost",
+            "mongo.local, test_1.my_hostmongo.local",
+            "blogs.mongodb.com, cluster.testmongodb.com"
+    })
+    @DisplayName("4. Throw when return address does not contain '.' separating shared part of domain")
+    void test4(final String srvHost, final String resolvedHost) {
+        doTest(srvHost, resolvedHost, true);
+    }
+
+    private void doTest(final String srvHost, final String resolvedHost, final boolean throwException) {
+        final ClusterId clusterId = new ClusterId();
+
+        final DnsResolver dnsResolver = new DefaultDnsResolver((name, type) -> singletonList(String.format("10 5 27017 %s",
+                resolvedHost)));
+
+        when(dnsSrvRecordMonitorFactory.create(eq(srvHost), eq(SRV_SERVICE_NAME), any(DnsSrvRecordInitializer.class))).thenAnswer(
+                invocation -> new DefaultDnsSrvRecordMonitor(srvHost, SRV_SERVICE_NAME, 10, 10,
+                            invocation.getArgument(2), clusterId, dnsResolver));
+
+        final ClusterSettings.Builder settingsBuilder = ClusterSettings.builder()
+                .mode(ClusterConnectionMode.MULTIPLE)
+                .requiredClusterType(ClusterType.SHARDED)
+                .srvHost(srvHost);
+
+        cluster = new DnsMultiServerCluster(clusterId, settingsBuilder.build(),
+                serverFactory,
+                dnsSrvRecordMonitorFactory);
+        try {
+            Thread.sleep(100); // racy
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            // ignore
+        }
```

Comment:
`ClusterFixture.sleep(100);`?

Reply:
the coding pattern in this class was mainly copied from another testing class, i.e. com.mongodb.internal.connection.SrvPollingProseTests.

I can use ClusterFixture.sleep(100) for sure.

but should we add `Thread.currentThread().interrupt()` in its code?